### PR TITLE
Remove irrelevant exceptions and update tests

### PR DESCRIPTION
**Pull request by qdpham13**
- Remove exceptions that users can't do anything about and that don't break Realtime
- Remote duplicate test and update test to not rely on exception

### CODE REVIEW

**Comment by danasilver** [comment]
Could you add some notes or code comments where the exceptions are removed/added indicating what the issue is/why we're catching and swallowing the exception?

File: firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java [hunk 1]
```diff
@@ -105,11 +105,8 @@ public void listenForNotifications() {
         handleNotifications(inputStream);
         inputStream.close();
       } catch (IOException ex) {
-        propagateErrors(
-            new FirebaseRemoteConfigClientException(
-                "Unable to parse config update message.",
-                ex.getCause(),
-                FirebaseRemoteConfigException.Code.CONFIG_UPDATE_MESSAGE_INVALID));
+        // Stream was interrupted due to a transient issue and the system will retry the connection.
+        Log.d(TAG, "Stream was cancelled due to an exception: " + ex.toString());
```
Comment (danasilver) [inline firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java:109]: Would include "Retrying" somewhere in this message.

Comment (qdpham13) [reply firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java]: added

File: firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java [hunk 2]
```diff
@@ -105,11 +105,12 @@ public void listenForNotifications() {
...
-        propagateErrors(
-            new FirebaseRemoteConfigClientException(
-                "Unable to parse config update message.",
-                ex.getCause(),
-                FirebaseRemoteConfigException.Code.CONFIG_UPDATE_MESSAGE_INVALID));
+        // Stream was interrupted due to a transient issue and the system will retry the connection.
+        Log.d(
+            TAG,
+            String.format(
+                "Stream was cancelled due to an exception: %s. Retrying the connection...",
```
Comment (danasilver) [inline firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java:112]: Sorry last nit 😅  Was thinking about the exception in the middle of the log and remembered `Log.d` can take an exception as a parameter and print it, so we can use those for the logs here: https://developer.android.com/reference/android/util/Log.html#d(java.lang.String,%20java.lang.String,%20java.lang.Throwable)

Comment (qdpham13) [reply firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigAutoFetch.java]: Added it to the logs

**Review by danasilver — APPROVED** [review]

**Pull request merged**